### ISSUE TITLE
There are multiple information leaks in cms-dev/cms

### ISSUE THREAD

**Issue by niuzhi** [cms-dev/cms#1160]
1.file: cms/cmscontrib/AddAdmin.py 
  function:add_admin
  Log information disclosure **username** and **password**
  
def add_admin(username, password=None):
    logger.info("Creating the admin on the database.")
    if password is None:
        password = generate_random_password()
    admin = Admin(username=username,
                  authentication=hash_password(password),
                  name=username,
                  permission_all=True)
    try:
        with SessionGen() as session:
            session.add(admin)
            session.commit()
    except IntegrityError:
        logger.error("An admin with the given username already exists.")
        return False

    logger.info("Admin with complete access added. "
                "Login with username %s and password %s",
                username, password)
    return True

**Comment by NhatMinh0208** [comment]
Uhmmmmmm...... that's normal. It's where the cmsAddAdmin script reminds you of the login info for the admin account you just created. That does not constitute a leak.

**Comment by Grackins** [comment]
I'm not sure about it but is it really normal to log plain text passwords?

In this special case I think it's ok since `logger.info` is being used as a normal stdout `print` statement.
Also there must be a way to let the user know the randomly generated password.

@niuzhi Do you have a more secure way in mind to do this?

**Comment by wil93** [comment]
I agree that we could at least log the password only when it gets randomly generated (i.e. the `password is None` case) and skip showing it when it's provided (i.e. when we call `cmsAddAdmin ... -p thepassword`)
